### ISSUE TITLE
Strange item_type found in p2p.log

### ISSUE THREAD

**Issue by abitmore** [cryptonomex/graphene#263]
`test1` tag.
"item_type":3624466160
"item_type":3624527728

```
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] beginning an iteration of advertise inventory
        node.cpp:1175
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] peer->peer_needs_sync_items_from_us: true
        node.cpp:1188
2015-08-22T22:20:33 p2p:advertise_inventory_loop  clear_old_inventory ] Expiring old inventory for peer 45.55.6.216:1776: removing 0 i
tems advertised to peer (0 left), and 0 advertised to us (0 left)                 peer_connection.cpp:479
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] peer->peer_needs_sync_items_from_us: false
        node.cpp:1188
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] inventory_to_advertise: [{"item_type":1001,"item_hash":"e53c5595bc7f37891ed00c277ddcef06e8eb37bd"}]                     node.cpp:1196
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] *peer->inventory_advertised_to_peer.find(item_to_advertise): {"item":{"item_type":3624466160,"item_hash":"ff7f00002b00000000000000ffffffffff7f0000"},"timestamp":"2023-11-27T10:29:02"}
                node.cpp:1200
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] *peer->inventory_peer_advertised_to_us.find(item_to_advertise): {"item":{"item_type":3624527728,"item_hash":"ff7f00000000000000000000b86309d8ff7f0000"},"timestamp":"1948-09-30T21:23:04"}
                node.cpp:1202
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] advertising item e53c5595bc7f37891ed00c277ddcef06e8eb37bd to peer 104.156.226.183:40479                 node.cpp:1212
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] advertising 1 new item(s) of 1 type(s) to peer 104.156.226.183:40479                    node.cpp:1218
2015-08-22T22:20:33 p2p:advertise_inventory_loop  clear_old_inventory ] Expiring old inventory for peer 104.156.226.183:40479: removing 0 items advertised to peer (105 left), and 0 advertised to us (0 left)                  peer_connection.cpp:479
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] peer->peer_needs_sync_items_from_us: true
        node.cpp:1188
2015-08-22T22:20:33 p2p:advertise_inventory_loop  clear_old_inventory ] Expiring old inventory for peer 127.0.0.1:62015: removing 0 items advertised to peer (0 left), and 0 advertised to us (27 left)                 peer_connection.cpp:479
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] peer->peer_needs_sync_items_from_us: false
        node.cpp:1188
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] inventory_to_advertise: [{"item_type":1001,"item_hash":"e53c5595bc7f37891ed00c277ddcef06e8eb37bd"}]                     node.cpp:1196
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] inventory_to_advertise: [{"item_type":1001,"item_hash":"e53c5595bc7f37891ed00c277ddcef06e8eb37bd"}]                     node.cpp:1196
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] *peer->inventory_advertised_to_peer.find(item_to_advertise): {"item":{"item_type":0,"item_hash":"000000002b00000000000000ffffffffff7f0000"},"timestamp":"2023-11-27T10:29:02"}                  node.cpp:1200
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] advertising 0 new item(s) of 0 type(s) to peer 178.62.88.151:53168                      node.cpp:1218
2015-08-22T22:20:33 p2p:advertise_inventory_loop  clear_old_inventory ] Expiring old inventory for peer 178.62.88.151:53168: removing 0 items advertised to peer (12 left), and 0 advertised to us (93 left)                    peer_connection.cpp:479
2015-08-22T22:20:33 p2p:advertise_inventory_loop advertise_inventory_ ] peer->peer_needs_sync_items_from_us: true
        node.cpp:1188
2015-08-22T22:20:33 p2p:advertise_inventory_loop  clear_old_inventory ] Expiring old inventory for peer 176.9.234.167:34762: removing 0 items advertised to peer (0 left), and 0 advertised to us (60 left)                     peer_connection.cpp:479
2015-08-22T22:20:33 p2p:advertise_inventory_loop         send_message ] peer_connection::send_message() enqueueing message of type 5001 for peer 104.156.226.183:40479                  peer_connection.cpp:365
2015-08-22T22:20:33 p2p:advertise_inventory_loop send_queueable_messa ] peer_connection::send_message() is firing up send_queued_message_task                   peer_connection.cpp:354
```

**Comment by theoreticalbts** [comment]
Good catch.  It's actually reading something undefined at the end of the vector.  The code reads like this:

```
          if (peer->inventory_advertised_to_peer.find(item_to_advertise) == peer->inventory_advertised_to_peer.end() )
             wdump((*peer->inventory_advertised_to_peer.find(item_to_advertise)));
          if (peer->inventory_peer_advertised_to_us.find(item_to_advertise) == peer->inventory_peer_advertised_to_us.end() )
             wdump((*peer->inventory_peer_advertised_to_us.find(item_to_advertise)));
```

These statements need fixed, they are currently dumping an undefined element is minimal value.  I'm not sure what they are intending to dump.  I am tempted to suggest removing these four lines entirely since we have enough p2p logspam.

I will assign this ticket to our p2p guru @emfrias but if I don't hear back in a day or two, I'm going to delete the code and close the ticket.

**Comment by theoreticalbts** [comment]
Fixed by 8ebc1cfc43e97060f19dae78fbaa3f361fd30aa8
